bokeh/embed.py

```diff
@@ -131,10 +132,37 @@ def components(models, wrap_script=True, wrap_plot_info=True):
             values.append(models[k])
         models = values
 
-    # 2) Do our rendering
-
-    with _ModelInDocument(models):
-        (docs_json, render_items) = _standalone_docs_json_and_render_items(models)
+    # 2) Append models to one document. Either pre-existing or new.
+    doc = None
+    for model in models:
+        if isinstance(model, Document):
+            doc = model
+            break
+        elif isinstance(model, Model):
+            if model.document is not None:
+                doc = model.document
+                break
+    if doc is None:
+        # oh well - just make up a doc
+        doc = Document()
+
+    _to_remove_after = []
+    for model in models:
+        if isinstance(model, Model):
+            if model.document is None:
+                try:
+                    doc.add_root(model)
+                except RuntimeError as e:
+                    child = re.search('\((.*)\)', str(e)).group(0)
+                    msg = ('Sub-model {0} of the root model {1} is already owned '
+                           'by another document (Models must be owned by only a '
+                           'single document). This may indicate a usage '
+                           'error.'.format(child, model))
+                    raise RuntimeError(msg)
+                _to_remove_after.append(model)
```

Comment:
This is a bit gorpy in the middle of an otherwise step-by-step function,  it might be nice to make a little helper function:
```
models_to_dedoc = _add_doc_to_models(...)
```